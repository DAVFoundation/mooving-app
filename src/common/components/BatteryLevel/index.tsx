import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { customColors, customSizes } from '../../styles';

interface IBatteryLevelProps {
  batteryLevel: number;
  style?: any;
  isDark?: boolean;
}

export default class BatteryLevel extends React.Component<IBatteryLevelProps> {
  constructor(props: IBatteryLevelProps) {
    super(props);
  }

  public render() {
    const { batteryLevel, style } = this.props;
    return (
      <View style={[style ? style : {}, styles.container]}>
        <Text
          style={[
            styles.chargePercentage,
            this.props.isDark ? styles.darkText : styles.lightText,
          ]}>
          {batteryLevel}%
        </Text>
        <View
          style={[
            styles.chargeIndicatorOuter,
            this.props.isDark ? styles.darkBorder : styles.lightBorder,
          ]}>
          <View
            style={[
              styles.chargeIndicatorInner,
              this.props.isDark ? styles.darkBackground : styles.lightBackground,
              { width: batteryLevel * .24 },
            ]}/>
          <View
            style={[
              styles.batteryTop,
              this.props.isDark ? styles.darkBorder : styles.lightBorder,
            ]}/>
        </View>
      </View>
    );
  }
}
const borderRadius = 4;

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    width: 75,
    paddingVertical: 5,
    paddingHorizontal: 5,
  },
  chargePercentage: {
    color: customColors.white,
    lineHeight: 15,
    marginRight: 5,
  },
  chargeIndicatorOuter: {
    borderRadius,
    borderWidth: 1,
    width: 30,
    height: 16,
    position: 'relative',
  },
  chargeIndicatorInner: {
    borderRadius: borderRadius / 2,
    borderWidth: 1,
    borderColor: 'transparent',
    margin: 2,
    width: 20,
    height: 10,
  },
  batteryTop: {
    borderTopRightRadius: borderRadius * 2,
    borderBottomRightRadius: borderRadius * 2,
    marginLeft: 3,
    width: 2,
    height: 6,
    borderWidth: 1,
    top: 4,
    right: -4,
    position: 'absolute',
  },
  darkText: {
    color: customColors.black,
  },
  lightText: {
    color: customColors.white,
  },
  darkBackground: {
    backgroundColor: customColors.black,
  },
  lightBackground: {
    backgroundColor: customColors.white,
  },
  darkBorder: {
    borderColor: customColors.blackTransparent,
  },
  lightBorder: {
    borderColor: customColors.whiteTransparent,
  },
});
