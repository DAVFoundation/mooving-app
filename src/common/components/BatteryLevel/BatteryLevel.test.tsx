import React from 'react';
import renderer from 'react-test-renderer';

import BatteryLevel from '../BatteryLevel';

it('renders correctly with defaults', () => {
  const view = renderer
    .create(<BatteryLevel
      batteryLevel={40}/>)
    .toJSON();
  expect(view).toMatchSnapshot();
});
