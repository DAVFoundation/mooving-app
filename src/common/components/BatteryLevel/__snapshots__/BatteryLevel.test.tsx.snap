// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingHorizontal": 5,
        "paddingVertical": 5,
        "width": 75,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "lineHeight": 15,
          "marginRight": 5,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    40
    %
  </Text>
  <View
    style={
      Array [
        Object {
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 16,
          "position": "relative",
          "width": 30,
        },
        Object {
          "borderColor": "rgba(255, 255, 255, 0.35)",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "transparent",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 10,
            "margin": 2,
            "width": 20,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
          Object {
            "width": 9.6,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderBottomRightRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 1,
            "height": 6,
            "marginLeft": 3,
            "position": "absolute",
            "right": -4,
            "top": 4,
            "width": 2,
          },
          Object {
            "borderColor": "rgba(255, 255, 255, 0.35)",
          },
        ]
      }
    />
  </View>
</View>
`;
